lib/maths/COutliers.cc

```diff
@@ -9,87 +9,514 @@
 #include <core/CDataFrame.h>
 
 #include <maths/CDataFrameUtils.h>
+#include <maths/CIntegration.h>
 #include <maths/CLinearAlgebraEigen.h>
 #include <maths/CTools.h>
 
-#include <boost/math/distributions/normal.hpp>
+#include <boost/math/distributions/lognormal.hpp>
 
+#include <numeric>
+#include <sstream>
 #include <tuple>
-#include <vector>
 
 namespace ml {
 namespace maths {
+namespace outliers_detail {
```

Comment:
Generally I always use ml::library::<filename>_detail for any namespace which is intended to include implementation details (there are quite a few different examples in the maths library following this convention). I quite like this because it immediately makes anyone aware that the things in the nested namespace are not really interface to functionality, but allows us to unit test them if we want.

The problem with not nesting inside maths is that it then needs to qualify all maths related functionality it uses. I also think there might be something to be said for down scoping some of the library namespaces, but I see this as part of the work to split the biggest libraries up somewhat. Given the current status I think I'd prefer to keep this as is.